### PR TITLE
Fix Ryzen Core Count

### PR DESCRIPTION
**Pull request by obelisk**
With AMD's new Ryzen processors, there is now a difference between core count and processor count. This adds a new column to the system table but it's not populated on other OS'. For Linux systems we report the number of logical cores as physical cores as well and it is the correct number.

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [1262](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1262/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [1263](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1263/)) failed one or more tests (**macOS/OS X**).

**Comment by jedi22** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [1904](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1904/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [1264](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1264/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [1265](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1265/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [5143](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5143/)) failed one or more tests (**Linux**).

**Comment by jedi22** [comment]
So interestingly, on Ubuntu 16.04 LTS, both physical and logical cores report 16 on a Ryzen 1800X

**Comment by jedi22** [comment]
/proc/cpuinfo has more information and lscpu can pull sockets, cores, and logical cores. Maybe we should add that as a wishlist.

**Comment by jedi22** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit c12cbaca83c4ecdbe58b5f12a9d0b64f5c715de2 (Job results: [1905](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1905/)) failed the code audit or documentation test.

**Comment by jedi22** [comment]
ok to test

File: osquery/tables/system/windows/system_info.cpp [hunk 1]
```diff
@@ -36,12 +36,16 @@ QueryData genSystemInfo(QueryContext& context) {
...
   WmiRequest wmiSystemReq("select * from Win32_ComputerSystem");
+  WmiRequest wmiSystemReqProc("select * from Win32_Processor");
   std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  std::vector<WmiResultItem>& wmiResultsProc = wmiSystemReqProc.results();
   if (wmiResults.size() != 0) {
     long numProcs = 0;
     wmiResults[0].GetLong("NumberOfLogicalProcessors", numProcs);
     r["cpu_logical_cores"] = INTEGER(numProcs);
     wmiResults[0].GetLong("NumberOfProcessors", numProcs);
+    r["cpu_physical_processors"] = INTEGER(numProcs);
```
Comment (poppyseedplehzr) [inline osquery/tables/system/windows/system_info.cpp]: Why are you adding in a new column? If you do this we lose compatibility with the other tables.. Is there any reason that this value is different than the `cpu_physical_cores`? If you add this here can you modify the other table schemas also?

File: specs/system_info.table [hunk 2]
```diff
@@ -6,6 +6,7 @@ schema([
     Column("cpu_type", TEXT, "CPU type"),
     Column("cpu_subtype", TEXT, "CPU subtype"),
     Column("cpu_brand", TEXT, "CPU brand string, contains vendor and model"),
+    Column("cpu_physical_processors", INTEGER, "Max number of CPU physical processors"),
```
Comment (theopolis) [inline specs/system_info.table]: Do you mean `sockets`? CPU ... processors is redundant.

File: osquery/tables/system/windows/system_info.cpp [hunk 3]
```diff
@@ -36,12 +36,14 @@ QueryData genSystemInfo(QueryContext& context) {
...
   WmiRequest wmiSystemReq("select * from Win32_ComputerSystem");
+  WmiRequest wmiSystemReqProc("select * from Win32_Processor");
   std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  std::vector<WmiResultItem>& wmiResultsProc = wmiSystemReqProc.results();
   if (wmiResults.size() != 0) {
     long numProcs = 0;
     wmiResults[0].GetLong("NumberOfLogicalProcessors", numProcs);
     r["cpu_logical_cores"] = INTEGER(numProcs);
-    wmiResults[0].GetLong("NumberOfProcessors", numProcs);
+    wmiResultsProc[0].GetLong("NumberOfCores", numProcs);
```
Comment (muffins) [inline osquery/tables/system/windows/system_info.cpp]: Ensure you check that `wmiSystemReqProc.empty()` before using.

Comment (obelisk) [reply osquery/tables/system/windows/system_info.cpp]: Good call!

**Comment by obelisk** [comment]
Running inside a Windows 10 VM with host Ubuntu 17.04 on a Ryzen 1800X
```
+-----------------+--------------------------------------+----------+-------------+-------------------------------------------------+--------------------+-------------------+-----------------+-----------------+----------------+------------------+-----------------+-----------------+-----------------+
| hostname        | uuid                                 | cpu_type | cpu_subtype | cpu_brand                                       | cpu_physical_cores | cpu_logical_cores | physical_memory | hardware_vendor | hardware_model | hardware_version | hardware_serial | computer_name   | local_hostname  |
+-----------------+--------------------------------------+----------+-------------+-------------------------------------------------+--------------------+-------------------+-----------------+-----------------+----------------+------------------+-----------------+-----------------+-----------------+
| DESKTOP-231S312 | E88B1799-EB21-7743-88C0-28953097F8F3 | x86_64   | -1          | AMD Ryzen 7 1800X Eight-Core Processor          | 6                  | 6                 | 4294496256      | innotek GmbH    | VirtualBox     | -1               | 0               | DESKTOP-231S312 | DESKTOP-231S312 |
+-----------------+--------------------------------------+----------+-------------+-------------------------------------------------+--------------------+-------------------+-----------------+-----------------+----------------+------------------+-----------------+-----------------+-----------------+
```

**Comment by muffins** [comment]
Looks good, merge when yer ready:
```
osquery> select hostname, cpu_brand, cpu_physical_cores, cpu_logical_cores from system_info;
+-----------+-----------------------------------------+--------------------+-------------------+
| hostname  | cpu_brand                               | cpu_physical_cores | cpu_logical_cores |
+-----------+-----------------------------------------+--------------------+-------------------+
| deathstar | Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz | 4                  | 4                 |
+-----------+-----------------------------------------+--------------------+-------------------+
```

**Pull request merged**